build/templates/_converters.py.mako

```diff
@@ -340,11 +340,14 @@ def ${enums[enum_name]['enum_to_converted_value_function_name']}(value):
 
 
 def ${enums[enum_name]['converted_value_to_enum_function_name']}(value):
-    return {
+    try:
+        return {
         % for enum_value in enums[enum_name]['values']:
-        ${helper.get_enum_value_snippet(enum_value['converts_to_value'])}: enums.${enums[enum_name]['python_name']}.${enum_value['python_name']},
+            ${helper.get_enum_value_snippet(enum_value['converts_to_value'])}: enums.${enums[enum_name]['python_name']}.${enum_value['python_name']},
         % endfor
-    }[value]
+        }[value]
+    except KeyError:
```

Comment:
I don't think it makes sense for function that goes from "converted value" to "enum" ro raise `ValueError` while the function that converts from "enum" to "converted value" raises `KeyError`.

They should both raise the same exception on unknown values.

This is mostly academic since calling code will do its own `except`+`raise` but might as well keep it consistent so nobody is confused about it.

I think `KeyError` is fine. It's simpler.

Reply:
As explained in https://github.com/ni/nimi-python/pull/1738#discussion_r905256369 and https://github.com/ni/nimi-python/pull/1738#discussion_r905261737, the calling code (i.e. the `AttributeEnumWithConverter class) has no context info about the supported values for conversion, so it cannot provide more detailed info after it catches the KeyError.

The data flows look like below:

**Getting attribute value:**
Driver runtime --> (convert to `Enum` with enum instantiation, which throws `ValueError`) --> Converter (which throws `KeyError`) --> User

**Setting attribute value:**
User --> Converter (which throws `KeyError`) --> (passes `enum,value` into C API) --> Driver runtime

If we want consistency across both converters, we can either:
1. Throw `ValueError` from both converters, then in the `AttributeEnumWithConverter` class:
    - Don't do anything for `__set__()` (let the `ValueError` with a list of valid values passes through to user)
    - Catch `ValueError` in `__get__()` and re-raise as `DriverTooNewError`
2. Throw `KeyError` from both converters, then in the `AttributeEnumWithConverter` class:
    - Catch `KeyError` in `__set__()` and re-raise as `ValueError` with generic error message
    - Catch `KeyError` and `ValueError` in `__get__()` and re-raise as `DriverTooNewError`

I'm going with option 2 (throwing `KeyError` from both converters now. Feel free to let me know if you prefer option 1.

Reply:
It makes sense to me to _raise_ KeyError from the converter functions. This is the simplest thing to do. Client code (in this case Attribute class) has more context and can raise something more suitable for the situation.